roxctl/central/debug/download_diagnostics_test.go

```diff
@@ -0,0 +1,105 @@
+package debug
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/common/flags"
+	io2 "github.com/stackrox/rox/roxctl/common/io"
+	"github.com/stackrox/rox/roxctl/common/printer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDownloadDiagnosticsTimeoutReached(t *testing.T) {
+	shutdownServer := make(chan struct{}, 1)
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		<-shutdownServer
+	}))
+	defer server.Close()
+	defer close(shutdownServer)
+
+	_, stdErr, err := executeDiagnosticsCommand(t, server.URL, 10*time.Millisecond, "")
+
+	require.Error(t, err)
+	assert.True(t, isTimeoutError(err))
+	assert.Contains(t, stdErr.String(), "Timeout has been reached while creating diagnostic bundle")
+}
+
+func TestDownloadDiagnosticsServerError(t *testing.T) {
+	expectedErrorStr := "test-server-error"
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.WriteHeader(http.StatusInternalServerError)
+		_, _ = rw.Write([]byte(expectedErrorStr))
+	}))
+	defer server.Close()
+
+	_, stdErr, err := executeDiagnosticsCommand(t, server.URL, 20*time.Second, "")
+
+	require.Error(t, err)
+	assert.False(t, isTimeoutError(err))
+	assert.Contains(t, err.Error(), expectedErrorStr)
+	assert.NotContains(t, stdErr.String(), "Timeout has been reached while creating diagnostic bundle")
+}
+
+func TestDownloadDiagnosticsSuccess(t *testing.T) {
+	zipFilename := "test-file.zip"
+	tempDir, errTempDir := os.MkdirTemp("", "prefix")
```

Comment:
nit: there's also [t.TempDir()](https://pkg.go.dev/testing#T.TempDir) that you can use. The advantage here would be that you don't need to do any "manual" cleanups.